nixos/modules/programs/qgroundcontrol.nix

```diff
@@ -0,0 +1,39 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.qgroundcontrol;
+in
+{
+
+  options = {
+    programs.qgroundcontrol = {
+      enable = lib.mkEnableOption "qgroundcontrol";
+
+      package = lib.mkPackageOption pkgs "qgroundcontrol" {};
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    # ModemManager is known to interfere with serial connections;
+    # qgroundcontrol recommends disabling it
+    systemd.services.ModemManager.enable = lib.mkDefault false;
```

Comment:
Completely nonblocking comment here, but I wonder if we could somehow better communicate this to a user? If a user decides to enable this service, NixOS would be completely silent about it. Maybe it would be worth adding a comment to the `enable` option, or maybe to the manual or wiki or something?

Reply:
The only use case I can think of where ModemManager and QGroundControl might be used simultaneously would be using a SIM card + data plan to have a portable ground control laptop.

I figured out you can tell ModemManager to ignore devices using udev rules. I'll replace this with a udev rule.